lib/sync/storage-manager-listener.js

```diff
@@ -0,0 +1,36 @@
+let cachedStore;
+let cachedListener;
+
+module.exports = function getStorageManagerListener(store) {
+  if (cachedListener && (!store || store === cachedStore)) { return cachedListener; }
+
+  if (!store || (typeof store.dispatch !== 'function')) { throw new Error('You must pass in a redux store the first time you call getStorageManagerListener'); }
+
+  cachedStore = store;
```

Comment:
If we happen to pass a different store in between invocations, does it make sense to set the `cachedStore` to the new passed in `store`?